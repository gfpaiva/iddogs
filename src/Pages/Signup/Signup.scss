@import '../../Scss/root';

.signup {
  &__icon {
    @include rfs(150);
    margin-bottom: 2rem;
  }
}
