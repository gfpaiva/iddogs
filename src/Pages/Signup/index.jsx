import React from 'react';
import PropTypes from 'prop-types';
import { Redirect } from 'react-router-dom';
import { FaDog } from 'react-icons/fa';

import Login from '../../Components/Login';
import { IsAutenticated } from '../../Utils/Auth';

import './Signup.scss';

export default function Signup({ checkAutenticated }) {
  if (checkAutenticated()) return <Redirect to="/feed/husky" />;

  return (
    <div className="signup text-center">
      <div className="signup__content">
        <FaDog className="signup__icon" />

        <p className="signup__welcome">
          Bem vindo ao&nbsp;
          <strong>IDDogs.</strong>
          <br />
          Uma galeria com fotos de cachorros filtrados pelas raças: husky, labrador, hound e pug.
        </p>

        <p className="signup__start">
          Para começar faça o login abaixo com o seu e-mail:
        </p>

        <Login />
      </div>
    </div>
  );
}

Signup.defaultProps = {
  checkAutenticated: IsAutenticated,
};

Signup.propTypes = {
  checkAutenticated: PropTypes.func,
};
