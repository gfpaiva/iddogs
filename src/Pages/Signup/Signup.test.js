import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { mount } from 'enzyme';

import Signup from './index';

describe('<Signup />', () => {
	it('should mount logged off properly', () => {
		const wrapper = mount(
      <Router>
        <Signup />
      </Router>
    );

		expect(wrapper.find('.signup')).toMatchSnapshot();
  });

	it('should mount logged in properly', () => {
    const checkAutenticated = jest.fn();
    checkAutenticated.mockReturnValue(true);

		const wrapper = mount(
      <Router>
        <Signup checkAutenticated={checkAutenticated} />
      </Router>
    );

		expect(wrapper.find('Redirect').prop('to')).toBe('/feed/husky');
	});
});
