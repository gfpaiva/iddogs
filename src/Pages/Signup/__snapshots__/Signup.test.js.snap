// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup /> should mount logged off properly 1`] = `
<div
  className="signup text-center"
>
  <div
    className="signup__content"
  >
    <FaDog
      className="signup__icon"
    >
      <IconBase
        attr={
          Object {
            "viewBox": "0 0 512 512",
          }
        }
        className="signup__icon"
      >
        <svg
          className="signup__icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M496 96h-64l-7.16-14.31A32 32 0 0 0 396.22 64H342.6l-27.28-27.28C305.23 26.64 288 33.78 288 48.03v149.84l128 45.71V208h32c35.35 0 64-28.65 64-64v-32c0-8.84-7.16-16-16-16zm-112 48c-8.84 0-16-7.16-16-16s7.16-16 16-16 16 7.16 16 16-7.16 16-16 16zM96 224c-17.64 0-32-14.36-32-32 0-17.67-14.33-32-32-32S0 174.33 0 192c0 41.66 26.83 76.85 64 90.1V496c0 8.84 7.16 16 16 16h64c8.84 0 16-7.16 16-16V384h160v112c0 8.84 7.16 16 16 16h64c8.84 0 16-7.16 16-16V277.55L266.05 224H96z"
            key="0"
          />
        </svg>
      </IconBase>
    </FaDog>
    <p
      className="signup__welcome"
    >
      Bem vindo ao 
      <strong>
        IDDogs.
      </strong>
      <br />
      Uma galeria com fotos de cachorros filtrados pelas raças: husky, labrador, hound e pug.
    </p>
    <p
      className="signup__start"
    >
      Para começar faça o login abaixo com o seu e-mail:
    </p>
    <withRouter(Login)>
      <Login
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        location={
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          }
        }
        match={
          Object {
            "isExact": true,
            "params": Object {},
            "path": "/",
            "url": "/",
          }
        }
      >
        <section
          className="login"
        >
          <Formik
            enableReinitialize={false}
            initialValues={
              Object {
                "email": "",
              }
            }
            isInitialValid={false}
            onSubmit={[Function]}
            validate={[Function]}
            validateOnBlur={true}
            validateOnChange={true}
          >
            <form
              className="login__form"
              onSubmit={[Function]}
            >
              <div
                className="login__input-container"
              >
                <StyledInput
                  autoComplete="off"
                  className="login__input"
                  error={false}
                  filled={false}
                  id="email"
                  name="email"
                  onChange={[Function]}
                  placeholder="Digite seu e-mail (ex.: your@email.com)"
                  required={true}
                  type="email"
                  value=""
                >
                  <label
                    className="styled-input"
                    htmlFor="email"
                  >
                    <input
                      autoComplete="off"
                      className="text-center login__input"
                      id="email"
                      name="email"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      placeholder="Digite seu e-mail (ex.: your@email.com)"
                      required={true}
                      type="email"
                      value=""
                    />
                  </label>
                </StyledInput>
              </div>
              <Button
                className="login__button my-2"
                disabled={false}
                small={false}
                type="submit"
              >
                <button
                  className="login__button my-2 button bg-secondary color-primary"
                  disabled={false}
                  type="submit"
                >
                  <span
                    className="button__content"
                  >
                    Entrar
                    <FaPaw
                      className="login__button-icon"
                    >
                      <IconBase
                        attr={
                          Object {
                            "viewBox": "0 0 512 512",
                          }
                        }
                        className="login__button-icon"
                      >
                        <svg
                          className="login__button-icon"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          strokeWidth="0"
                          style={
                            Object {
                              "color": undefined,
                            }
                          }
                          viewBox="0 0 512 512"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M256 224c-79.41 0-192 122.76-192 200.25 0 34.9 26.81 55.75 71.74 55.75 48.84 0 81.09-25.08 120.26-25.08 39.51 0 71.85 25.08 120.26 25.08 44.93 0 71.74-20.85 71.74-55.75C448 346.76 335.41 224 256 224zm-147.28-12.61c-10.4-34.65-42.44-57.09-71.56-50.13-29.12 6.96-44.29 40.69-33.89 75.34 10.4 34.65 42.44 57.09 71.56 50.13 29.12-6.96 44.29-40.69 33.89-75.34zm84.72-20.78c30.94-8.14 46.42-49.94 34.58-93.36s-46.52-72.01-77.46-63.87-46.42 49.94-34.58 93.36c11.84 43.42 46.53 72.02 77.46 63.87zm281.39-29.34c-29.12-6.96-61.15 15.48-71.56 50.13-10.4 34.65 4.77 68.38 33.89 75.34 29.12 6.96 61.15-15.48 71.56-50.13 10.4-34.65-4.77-68.38-33.89-75.34zm-156.27 29.34c30.94 8.14 65.62-20.45 77.46-63.87 11.84-43.42-3.64-85.21-34.58-93.36s-65.62 20.45-77.46 63.87c-11.84 43.42 3.64 85.22 34.58 93.36z"
                            key="0"
                          />
                        </svg>
                      </IconBase>
                    </FaPaw>
                  </span>
                </button>
              </Button>
            </form>
          </Formik>
        </section>
      </Login>
    </withRouter(Login)>
  </div>
</div>
`;
